### PR TITLE
New cli tool to control task association configs

### PR DESCRIPTION
**Pull request by boajer**
https://oat-sa.atlassian.net/browse/NEX-1135

Helper cli scipt to add association mapping for the task brokers

May consider support of: 

- [ ] removing  associations
- [ ] killing and cleaning queues

### CODE REVIEW

File: scripts/tools/ManageAssociationMap.php [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    protected function provideUsage()
+    {
+        return [
+            'prefix' => 'h',
+            'longPrefix' => 'help',
+            'description' => 'Prints the help.'
+        ];
+    }
+
+    protected function run()
```
Comment (gabrielfs7) [inline scripts/tools/ManageAssociationMap.php:72]: I did not test / used it before:

1) I think we could at least extract the logic from `run()` to methods (or classes) with meaningful names. 
2) Also as it is a new important feature, should be add to the README.md how/why to use it.
3) Can we add some tests, please (after refactoring, tests should be smaller an easier).

Comment (bartlomiejmarszal) [reply scripts/tools/ManageAssociationMap.php:72]: https://github.com/oat-sa/extension-tao-task-queue/pull/117/commits/fbf7e0057f665af412898f69e0aeb14c4ccd7dcc

File: scripts/tools/ManageAssociationMap.php [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        $this->getQueueDispatcher()->setOptions($existingOptions);
+
+        $this->getServiceManager()->register(QueueDispatcherInterface::SERVICE_ID, $this->getQueueDispatcher());
+
+        $initializer = new InitializeQueue();
+        $this->propagate($initializer);
+        return $initializer;
+    }
+
+    protected function getQueueDispatcher(): QueueDispatcher
```
Comment (gabrielfs7) [inline scripts/tools/ManageAssociationMap.php]:
```suggestion
    private function getQueueDispatcher(): QueueDispatcher
```
Can it be private?

File: scripts/tools/ManageAssociationMap.php [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    {
+        return $this->getServiceLocator()->get(QueueDispatcher::SERVICE_ID);
+    }
+
```
Comment (gabrielfs7) [inline scripts/tools/ManageAssociationMap.php]:
```suggestion
```

File: model/Service/QueueAssociationService.php [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+use oat\oatbox\service\ConfigurableService;
+use oat\tao\model\taskQueue\Queue;
+use oat\tao\model\taskQueue\QueueDispatcher;
+use oat\tao\model\taskQueue\QueueDispatcherInterface;
+use oat\taoTaskQueue\model\QueueBroker\RdsQueueBroker;
+use oat\taoTaskQueue\scripts\tools\InitializeQueue;
+
+class QueueAssociationService extends ConfigurableService
+{
+    public function addTaskQueueAssociations(string $taskClass, string $queue): InitializeQueue
```
Comment (gabrielfs7) [inline model/Service/QueueAssociationService.php]:
```suggestion
    public function associate(string $taskClass, string $queue): InitializeQueue
```
The class + namespace already provide the context IMO

**Comment by gabrielfs7** [comment]
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by gabrielfs7 — APPROVED** [review]
LGTM. Ok for the code. Just having issues testing locally, but I saw it working on the demo

**Review by mike-ionut-mihai-sandu-tao — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**